src/main/java/org/sagebionetworks/web/client/widget/discussion/modal/NewDiscussionThreadModalViewImpl.java

```diff
@@ -103,4 +106,19 @@ public void setAlert(Widget w) {
 	public void showSuccess() {
 		DisplayUtils.showInfo(SUCCESS_TITLE, SUCCESS_MESSAGE);
 	}
+
+	@Override
+	public void setSendingRequestVisible(boolean visible) {
+		sendingRequest.setVisible(visible);
+	}
+
+	@Override
+	public void setSaveButtonEnabled(boolean enabled) {
+		saveButton.setEnabled(enabled);
```

Comment:
Use saveButton.state().loading() (and saveButton.state().reset()), and set the dataLoadingText button parameter in the ui xml.